light_group_mixer/light_group_mixer.py

```diff
@@ -0,0 +1,244 @@
+import nuke
+
+
+def update_light_group_mixer() -> None:
+    """This function adds/removes our lightgroups to/from our gizmo."""
+    lightgroup_mixer_node = nuke.thisNode()
+
+    channels = _get_all_channels(lightgroup_mixer_node)
+    if channels is None:
+        return
+
+    all_lightgroups, lightgroups_to_build = _get_lightgroups(
+        lightgroup_mixer_node, channels
+    )
+
+    to_connect_to = _get_node_to_connect_to()
+
+    _add_lightgroups(lightgroups_to_build, to_connect_to, lightgroup_mixer_node)
+
+    lightgroups_to_remove = _get_old_unused_lightgroups(
+        lightgroup_mixer_node, all_lightgroups
+    )
+    _remove_lightgroupgrade_nodes(lightgroup_mixer_node, lightgroups_to_remove)
+
+    reset_solos()
+
+
+def _get_all_channels(lightgroup_mixer_node: nuke.Node) -> list:
+    """This function returns all the channels from the input node."""
+    try:
+        return lightgroup_mixer_node.input(0).channels()
+    except AttributeError:
+        nuke.message("You haven't connected your input node.")
+        return None
+
+
+def _get_node_to_connect_to() -> nuke.Node:
```

Comment:
Is this function called within the gizmo/group (if its within a gizmo, definitely change it to a group as gizmo's cant store what's changed inside)